### PR TITLE
feat: warn when a getter conflicts with the state

### LINKED ISSUE #1331: Warn when Getter with same name as State
### Reproduction

https://github.com/TorchlightSoftware/nuxt-bugs/tree/pinia-getter-state-conflict

### Steps to reproduce the bug

Create a getter with the same name as a property in the state.

```ts
import {defineStore} from 'pinia'

export const useAuthStore = defineStore('auth', {
  state: () => ({
    user: {fname: 'Donna', lname: 'Renolds'},
  }),

  getters: {
    user: (state) => state.user,
  },

  actions: {
  },
})

```

### Expected behavior

This should produce an error telling the application developer not to do this.

### Actual behavior

When the value is consumed, it overflows the stack:

```
[nuxt] [request error] Maximum call stack size exceeded
  at formatRaw (node:internal/util/inspect:1037:12)  
  at formatValue (node:internal/util/inspect:817:10)  
  at inspect (node:internal/util/inspect:347:10)  
  at formatWithOptionsInternal (node:internal/util/inspect:2167:40)  
  at formatWithOptions (node:internal/util/inspect:2029:10)  
  at console.value (node:internal/console/constructor:324:14)  
  at console.log (node:internal/console/constructor:360:61)  
  at Proxy.me (./.nuxt/dist/server/server.mjs:2956:15)  
  at ReactiveEffect.fn (./node_modules/pinia/dist/pinia.mjs:1159:38)  
  at ReactiveEffect.run (./node_modules/@vue/reactivity/dist/reactivity.cjs.js:168:25)
```

### Additional information

Beginner mistake, not understanding that the getter is unnecessary in this case.  A clear error message would save time and help to more quickly determine the cause.  Thanks!

### PR DESCRIPTION
**Pull request by BlackCrowxyz**
### Description:
This allows Pinia store to warn user if they are using the same getter and state property name in a store.

### Related Issue:
close #1331 

### Testing Notes:
Passed all tests, didn't add any new test

### Side Note:
I think showing the message using `consloe.error` is more understandable than `consloe.warn`. Because in some cases equal names leads to furthur errors.

1. using `consloe.error`:

photo_2022-06-10_12-52-10

2. using `consloe.warn`:
photo_2022-06-10_12-52-12

### CODE REVIEW

File: packages/pinia/src/store.ts [hunk 1]
```diff
@@ -149,6 +149,17 @@ function createOptionsStore<
       localState,
       actions,
       Object.keys(getters || {}).reduce((computedGetters, name) => {
+        if (__DEV__) {
+          Object.keys(pinia.state.value[id] || {}).forEach((stateName) => {
```
Comment (posva) [inline packages/pinia/src/store.ts]: Let's check if the name exists in `localState`, it will be much faster

File: packages/pinia/src/store.ts [hunk 2]
```diff
@@ -149,6 +149,17 @@ function createOptionsStore<
...
+            if (name === stateName) {
+              console.error(
```
Comment (posva) [inline packages/pinia/src/store.ts]:
```suggestion
              console.warn(
```

File: packages/pinia/src/store.ts [hunk 3]
```diff
@@ -149,6 +149,17 @@ function createOptionsStore<
...
+                `[🍍]: "state" property name cannot be the same as "getters" property name.\n` +
+                  `\tGetter "${name}" is the same as state "${stateName}" in "${id}" store.`
```
Comment (posva) [inline packages/pinia/src/store.ts]:
```suggestion
                `[🍍]: A getter cannot have the same name as another state property. Rename one of them. Found with "${name}" in store "${id}".`
```

**Review by posva — CHANGES_REQUESTED** [review]
Can you add a test? There are other tests for warns so it should be easy to copy one.

File: packages/pinia/__tests__/store.spec.ts [hunk 4]
```diff
@@ -366,4 +366,17 @@ describe('Store', () => {
     useMyStore()
     expect(warnTextCheckPlainObject('poInit')).toHaveBeenWarnedTimes(0)
   })
+
+  it('only warns when state name conflicts with getters name', () => {
+    const useStore = defineStore({
+      id: 'main',
```
Comment (posva) [inline packages/pinia/__tests__/store.spec.ts]: Pass it as the first argument instead

File: packages/pinia/src/store.ts [hunk 5]
```diff
@@ -150,7 +150,7 @@ function createOptionsStore<
       actions,
       Object.keys(getters || {}).reduce((computedGetters, name) => {
         if (__DEV__) {
-          Object.keys(pinia.state.value[id] || {}).forEach((stateName) => {
+          Object.keys(localState || {}).forEach((stateName) => {
```
Comment (posva) [inline packages/pinia/src/store.ts]: No, but just use the in operator instead of forEach 😅

File: .vscode/settings.json [hunk 6]
```diff
@@ -3,7 +3,8 @@
     "editor.formatOnSave": true
   },
   "[typescript]": {
-    "editor.formatOnSave": true
+    "editor.formatOnSave": true,
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
```
Comment (posva) [inline .vscode/settings.json]: Remove this please

**Review by posva — CHANGES_REQUESTED** [review]

File: packages/pinia/src/store.ts [hunk 7]
```diff
@@ -149,14 +149,10 @@ function createOptionsStore<
...
       Object.keys(getters || {}).reduce((computedGetters, name) => {
-        if (__DEV__) {
-          Object.keys(localState || {}).forEach((stateName) => {
-            if (name === stateName) {
-              console.warn(
-                `[🍍]: A getter cannot have the same name as another state property. Rename one of them. Found with "${name}" in store "${id}".`
-              )
-            }
-          })
+        if (__DEV__ && name in (localState || {})) {
```
Comment (posva) [inline packages/pinia/src/store.ts]: do we really need the `|| {}` check?

```suggestion
        if (__DEV__ && name in localState) {
```

**Review by posva — APPROVED** [review]

File: packages/pinia/__tests__/store.spec.ts (hunk 4 above)
Comment (BlackCrowxyz) [reply packages/pinia/__tests__/store.spec.ts]: What is the difference between passing store name(id) as the first argument and using it as property?
Other tests weren't using id as the first argument!

**Pull request merged**